### DISCUSSION TITLE
"Pulsating" Brightness on SH1106 1.3" SPI Display

### DISCUSSION THREAD

**Discussion by Fishbone69** [olikraus/u8g2#2131] [Q&A]
Hey all.  My project is coming along pretty well.  I swapped out my DS3231 I2C RTC module for a DS3234 SPI RTC module.  I figured it would save me a pin since my SH1106 display is also SPI and I could put them all on the same bus.  Initially, it didn't work at all becasue I had to initialize the SPI bus for use with the DS3234.  I used :
` SPI.begin(2,0,4,10); // Initialize SPI for DS3234'
so pin2 is the SCLK, pin 0 = MISO, pin4= MOSI and pin 10 =CS.  
When I defined that, everything seemed to work except when I would make changes and upload my sketch, the screen would pulsate alternating between dim and bright at 1-2 Hz roughly.  If I unplug power from the USB and then plug it back in, the pulsating goes away.  I was wondering if you guys have seen anything like that.  I have a couple of hunches but not sure.
1.  In another post I noted how I connected by reset pin to +3.3V rather than some other pin.  I'm wondering if that was o.k. with the I2C setup, but now that the RTC is on the same SPI bus as the display, maybe I need to use the RC circuit.
2. The other thought I had is it is just the oled and RTC not playing nice together on the SPI bus.  That would suck if that is the case, because I spent about 3 days re-writing some library files for the DS3234.  
Anyway, let me know if you guys have seen this "strobing " effect and have any ideas.

Fish

**Comment by olikraus** [comment]
To me it sounds like a power supply issue. Maybe it helps if you add a 1uF capacitor very close to the power supply pins of the OLED.

**Reply by Fishbone69** [reply]
It is the reset pin to +3.3V (see this https://github.com/olikraus/u8g2/discussions/2124).  I put it on a proper gpio and it goes away, as soon as I tie it straight to +3.3V it comes back.  I will create the RC circuit as described in the earlier post and see if that helps.  If not, I'll try a voltage supervisor.

**Reply by Fishbone69** [reply]
o/k..  RC circuit didn't work.  I tried up to a 2.2uF and it didn't help.  I also tried to clean up the voltage around the display power supply with some 2.2uF caps...nothing.  Seems like the only thing so far that works consistently is the gpio pin to reset.  It is interesting that this didn't happen with the same display but using an I2C RTC.  Switching to an SPI RTC was going to save me a gpio pin, I may just go back and run the other config.  I know nothing about voltage supervisors...will look into that.  Thanks Folks!

**Comment by Fishbone69** [comment]
Oli, you know what bothers me about this?  It seems like it is something that you should be able to fix with software or minimal hardware - like the RC circuit.  Why that circuit didn't work...I dunno.  seems like it should.  Some mentioned on that post that under brownout, it wouldn't work.  And why is it not an issue when connecting an I2C device?  We know it is much slower...could that delay be helping?  I tried the RC circuit again with a 10uF cap.  It appeared to work...booted up 3 times in a row with no dimming, but then I started to get the issue.  It has never happened with an assigned reset pin.  Does U8g2 do anything with the reset pin?  II do not have a pullup resistor nor assign it an internal pullup in my code.  In fact, I do not initialize the pin at all in my code...just provide the pin number in the u8g2 constructor.  Given that,  what controls the voltage to the pin i.e. provides the logic for the reset and then tells it to go to +3.3V?

Fish

**Reply by olikraus** [reply]
U8g2 pulls the reset low and keeps it low for a configured time for the SH1106. U8g2 just uses the standard Arduino functions for this.
The I2C interface also would require a reset circuit. I do not know why the reset input is not there for many I2C OLEDs.

There is one issue by the way: The delay caused by the RC circuit must be considered in your code, i mean you must wait to call u8g2.begin() until the reset is done by the RC circut. The delay in your code should be
```
6*R*C 
```
before you call u8g2.begin().
So if you have 10kOhm and 10uF it is:
```
6*10000*0.00001 = 0.6 seconds = 600 milliseconds.
```
So you would require a delay(600) before u8g2.begin() to be on the safe side.

**Reply by Fishbone69** [reply]
i'll give it a try

**Reply by Fishbone69** [reply]
I tried it and it didn't help...but based on what you said, I think there may be something else.  This only happens after I upload changes to the sketch.  If I unplug the usb then plug it back it...it works fine....even when connected to +3.3V.  So when I upload a new sketch and the program restarts, does the power cycle?

**Reply by olikraus** [reply]
Probably not
